src/link/Dwarf.zig

```diff
@@ -560,6 +561,222 @@ pub const DeclState = struct {
             },
         }
     }
+
+    pub fn genArgDbgInfo(
+        self: *DeclState,
+        name: [:0]const u8,
+        ty: Type,
+        atom: *Atom,
+        loc: union(enum) {
+            register: u8,
+            stack: struct { fp_register: u8, offset: i32 },
```

Comment:
For the future we should also consider using the stack pointer as the base register (in order to relieve register pressure when it's ok to omit the frame pointer)

Reply:
Yep, totally agree. There is also a CLI flag to that effect we will need to respect `-fno-frame-pointer`.

Reply:
I'm only aware of `-fomit-frame-pointer`. As far as I understand, this only tells the compiler that it is _possible_ to omit the frame pointer, not a rigid requirement.

```
joachim@dustbowl ~> zig build-exe --help | grep frame-pointer
  -fomit-frame-pointer      Omit the stack frame pointer
  -fno-omit-frame-pointer   Store the stack frame pointer
```

Reply:
Oops, that's what I meant yes.